### PR TITLE
Fixed clk clock control driver

### PR DESCRIPTION
**Pull request by mfischer**

### CODE REVIEW

**Comment by mfischer** [comment]
@erwango FYI

**Review by bluerise — APPROVED** [review]

File: drivers/clock_control/Kconfig.fixed [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Fixed clock control driver config
+
+# Copyright (c) 2022 Google, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+config CLOCK_CONTROL_FIXED_RATE_CLOCK
+	bool "Fixed Clock Clock Control"
+	default n
+	help
+	  Enable driver for device-tree defined fixed clocks.
```
Comment (henrikbrixandersen) [inline drivers/clock_control/Kconfig.fixed:10]:
```suggestion
	  Enable driver for devicetree defined fixed clocks.
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Minor change suggestion, otherwise looks good.

I had a talk with @erwango at ZDS regarding the overhead introduced by adding such a driver, especially for system configurations with many fixed clocks. One solution could be introducing a common "clock manager" for reducing the overhead, but I think such an addition lies too far ahead in time for it to block a simple driver as proposed here.

I will try to sketch up our ideas for a clock manager and post them as an RFC.

File: drivers/clock_control/Kconfig.fixed (hunk 1 above)
Comment (mfischer) [reply drivers/clock_control/Kconfig.fixed]: Done.

**Comment by mfischer** [comment]
> Minor change suggestion, otherwise looks good.
> 

Thanks, done.

> I had a talk with @erwango at ZDS regarding the overhead introduced by adding such a driver, especially for system configurations with many fixed clocks. One solution could be introducing a common "clock manager" for reducing the overhead, but I think such an addition lies too far ahead in time for it to block a simple driver as proposed here.

Yeah I had a quick chat with him, too. I think we can solve those cases by having a 'bigger' clock driver to encapsulate the SoC behavior on a per SoC basis (e.g. for things such as clk muxes).

> I will try to sketch up our ideas for a clock manager and post them as an RFC.

Excellent. I'll be on the look-out. If we can generalize the bigger per-SoC driver with a generic binding, that'd be excellent.

I'll look at putting the code-gen/ API part into an RFC of itself and we can discuss there about how to best do that part.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by hongshui3000** [comment]
> I will try to sketch up our ideas for a clock manager and post them as an RFC.

Where can I see RFC? I am also very interested in this RFC, and I also hope that the current clock architecture can be enhanced in this direction

**Comment by henrikbrixandersen** [comment]
> Where can I see RFC? I am also very interested in this RFC, and I also hope that the current clock architecture can be enhanced in this direction

Sorry, it's not up yet.

**Pull request merged**